src/enterdb.erl

```diff
@@ -529,6 +530,55 @@ read_range_n_ts_(Tab, TD, {ok, DBKey, HashKey}, N, false) ->
 read_range_n_ts_(_Tab, _TD, {error, _} = E, _N, _) ->
     E.
 
+%%--------------------------------------------------------------------
+%% @doc
+%% Reads N number of Keys from table time_series table
+%% with name Name starting form StartKey.
+%% Stop if we hit StopKey before N number of results are returned.
+%% @end
+%%--------------------------------------------------------------------
+-spec read_range_n_ts(Name :: string(),
+		      StartKey :: key(),
+		      StopTs  :: {string(), term()},
+		   N :: pos_integer()) ->
+    {ok, [kvp()]} | {error, Reason :: term()}.
+read_range_n_ts(Name, StartKey, StopTs, N) ->
```

Comment:
Why do we need this when there is read_range/3? Or can't we add a new function called read_range_ts/3?

Reply:
I don't really like the current name.
But the differences are read_range/3 and read_rang_n/3 operates on all shards. Taking resources on each shard. 
While when using TimeSeries data, we can dive directly to the correct shard.
And do just keep doing continuous reading until we hit N or stop TS.
Read performance while doing writes are quite good.

But I think the API could be nicer.

Reply:
I see now.
You do;
SKey = lists:keyreplace("ts", 1, StartKey, StopTs),
Which is not so generic. I think read_range/3 should be the base for this function also. New fun name can be read_range_local/3. 
Give a start key and stop key as range. Changing parameter can be ts or any hash_excluded parameter but it is defined in table options anyway.
Using Chunks instead of N is more intuitive. You read chunk by chunk until you reach stop key.
Return error if start and stop keys do not end up in same shard.

Then other function is read_range_n_local/3. 
Give start key, number of entries(since it is enter db :) ) = N, and again you know what field to be excluded in hash already.

What do you say?

Reply:
Making it more generic sounds good. I think I like the name read_range_local better than the ts one as well.
And the stop key can be {Fieldname, Field} or full key if you want to provide it.

The reason for having both N and stop is since we don't know how much data there is the DB.
We don't want to read out 1M records between 1pm and 2pm if there are that many log entires during that time.
Then we will get a continuation and the first N entries. 
Same goes if we ask for 1k entries but there are only one. then the result will not be beyond 2pm. And state complete data set for the prefix/local/hashed part of key.

I think I will set a limit on max entries we are reading out as well.

"ts" is treated special already when enabling time series table. but that is ok I think.
-spec get_hash_key_def(KeyDef :: [string()],
		       HashExclude :: [string()],
		       TimeSeries :: boolean()) ->
    HashKey :: [string()].
get_hash_key_def(KeyDef, HashExclude, false) ->
    build_hash_key_def(KeyDef, HashExclude, []);
get_hash_key_def(KeyDef, HashExclude, true) ->
    build_hash_key_def(KeyDef, ["ts" | HashExclude], []).

Reply:
read_range_n_local/3 is there already but named read_range_n_ts/3. 
It is generic even if the name states otherwise. It just takes hashed part of key as prefix.
Will align the name of that one with the /4 where a stop key are provided as well.